src/server/pps/server/master.go

```diff
@@ -403,3 +385,209 @@ func (a *apiServer) deleteWorkersForPipeline(pipelineInfo *pps.PipelineInfo) err
+				return backoff.RetryNotify(func() error {
+					return a.makeCronCommits(pachClient, in)
+				}, backoff.NewInfiniteBackOff(), notifyCtx(pachClient.Ctx(), "cron for "+in.Cron.Name))
+			})
+		}
+	})
+	if !pipelineInfo.Standby {
+		// Standby is false so simply put it in RUNNING and leave it there.  This is
+		// only done with eg.Go so that we can handle all the errors in the
+		// same way below, it should be a very quick operation so there's no
+		// good reason to do it concurrently.
+		eg.Go(func() error {
+			return backoff.RetryNotify(func() error {
+				return setPipelineState(pps.PipelineState_PIPELINE_RUNNING)
+			}, backoff.NewInfiniteBackOff(), notifyCtx(pachClient.Ctx(), "set running (Standby = false)"))
+		})
+	} else {
+		// Capacity 1 gives us a bit of buffer so we don't needlessly go into
+		// standby when SubscribeCommit takes too long to return.
+		ciChan := make(chan *pfs.CommitInfo, 1)
+		eg.Go(func() error {
+			return backoff.RetryNotify(func() error {
+				return pachClient.SubscribeCommitF(pipelineInfo.Pipeline.Name, pipelineInfo.OutputBranch, "", pfs.CommitState_READY, func(ci *pfs.CommitInfo) error {
+					ciChan <- ci
+					return nil
+				})
+			}, backoff.NewInfiniteBackOff(), notifyCtx(pachClient.Ctx(), "SubscribeCommit"))
+		})
+		eg.Go(func() error {
+			return backoff.RetryNotify(func() error {
+				if err := setPipelineState(pps.PipelineState_PIPELINE_STANDBY); err != nil {
+					return err
+				}
+				for {
+					var ci *pfs.CommitInfo
+					select {
+					case ci = <-ciChan:
+						if ci.Finished != nil {
+							continue
+						}
+
+						if err := setPipelineState(pps.PipelineState_PIPELINE_RUNNING); err != nil {
+							return err
+						}
+
+						// Stay running while commits are available
+					running:
+						for {
+							// Wait for the commit to be finished before blocking on the
+							// job because the job may not exist yet.
+							if _, err := pachClient.BlockCommit(ci.Commit.Repo.Name, ci.Commit.ID); err != nil {
+								return err
+							}
+							if _, err := pachClient.InspectJobOutputCommit(ci.Commit.Repo.Name, ci.Commit.ID, true); err != nil {
+								return err
+							}
+
+							select {
+							case ci = <-ciChan:
+							default:
+								break running
+							}
+						}
+
+						if err := setPipelineState(pps.PipelineState_PIPELINE_STANDBY); err != nil {
+							return err
+						}
+					case <-pachClient.Ctx().Done():
+						return context.DeadlineExceeded
+					}
+				}
+			}, backoff.NewInfiniteBackOff(), func(err error, d time.Duration) error {
+				select {
+				case <-pachClient.Ctx().Done():
+					return context.DeadlineExceeded
+				default:
+					fmt.Printf("error in monitorPipeline: %v: retrying in: %v\n", err, d)
+				}
+				return nil
+			})
+		})
+	}
+	if err := eg.Wait(); err != nil {
+		fmt.Printf("error in monitorPipeline: %v", err)
+	}
+}
+
+// makeCronCommits makes commits to a single cron input's repo. It's
+// a helper function called by jobSpawner
```

Comment:
Might want to change the comment since it is in monitor pipeline now.

Reply:
Done